File: inheritance.cpp
#include<bits/stdc++.h>
using namespace std;

class Animal{
    public:
    string name;
    float weight;

    void Eat(){
        cout<<name<<" can Eat."<<endl;
    }
};

class Cat:public Animal{
    public:
        bool sound;

        Cat(string name, float weight, bool sound){
            this->name = name;
            this->weight = weight;
            this->sound = sound;
        }
};

int main(){
    Cat kitty("Kitty", 12.5, true);

    cout<<kitty.name<<" "<<kitty.weight<<" "<<kitty.sound<<endl;
    kitty.Eat();

    return 0;
}

File: inheritance.py

class Animal:
    name = ""
    weight = 0

    def Eat(self):
        print(f"{self.name} can eat")
    

class Cat(Animal):
    sound = False

    def __init__(self, name, weight, sound):
        self.name = name
        self.weight = weight
        self.sound = sound



kitty = Cat("Kitty", 12.5, True)

print(kitty)
print(kitty.name, kitty.weight, kitty.sound)
kitty.Eat()